### PR TITLE
1032 move gcc puller tables to airflow variable

### LINKED ISSUE #1032: Move gcc puller tables to Airflow variable

### PR DESCRIPTION
**Pull request by gabrielwol**
## What this pull request accomplishes:

- move GCC layers and Primary key dictionaries to Airflow variables
- Dynamic DAG generation for bigdata/ptc GCC layers using Airflow variable which stores credentials, deployment
- added mat view refreshes to Airflow variable: easier addition of new aggregations!

## Issue(s) this solves:

- Closes #1032 

## What, in particular, needs to reviewed:

- I couldn't think of an obvious way to make mat view aggregations their own Airflow tasks in a dynamic way so I just added them to the pull_layers task. Thoughts?

## What needs to be done by a sysadmin after this PR is merged
- change data_scripts link in Morbius

### CODE REVIEW

**Review by chmnata — CHANGES_REQUESTED** [review]
Can we make the mapped task's name as the layer's name? for refreshing the materialized view, I'm thinking we could use a trigger function on the table instead 🤔

File: dags/gcc_layers_pull.py [hunk 1]
```diff
@@ -17,115 +17,83 @@
...
-    'start_date': pendulum.datetime(2022, 11, 3, tz="America/Toronto"),
-    'email_on_failure': False, 
-    'retries': 0,
-    'on_failure_callback': partial(task_fail_slack_alert, use_proxy=True)
-}
+            #refresh mat views as necessary
+            agg_sql = layer[1].get("agg")
+            if agg_sql is not None:
+                with conn.cursor() as cur:
+                    print(agg_sql)
```
Comment (chmnata) [inline dags/gcc_layers_pull.py:59]: this for logging purpose?

**Review by chmnata — APPROVED** [review]
Thanks for reminding me that morbius's airflow version doesn't have mapped tasks naming yet :meow_headache: This looks good! refreshing mat view as a tasks as good as well.

File: dags/gcc_layers_pull.py (hunk 1 above)
Comment (gabrielwol) [reply dags/gcc_layers_pull.py:59]: Oh I'll remove this, it was for testing (instead of execute).

**Pull request merged**